### PR TITLE
Add logging and assertions, and refactor constants in Expense Tracker

### PR DESCRIPTION
**Pull request by YitHien**

### CODE REVIEW

**Comment by mathanmahe** [comment]
LGTM!

**Comment by YitHien** [comment]
Fixes #41

File: src/main/java/seedu/allonus/expense/ExpenseParser.java [hunk 1]
```diff
@@ -6,10 +6,26 @@
...
     public static final String ALL_DELIMITERS = "[dacr]/";
+    public static final String ASSERT_INPUT_NOT_NULL = "User input should not be null";
+    public static final String ASSERT_DELIMITER_NOT_NULL = "Delimiter should not be null";
+    public static final String ASSERT_RESULT_NOT_NULL = "Result should not be null";
+    public static final int EXPENSE_FIELDS = 1;
+    public static final int SPLIT_IN_HALF = 2;
+    public static final int RIGHT_SIDE = 1;
+    public static final int LEFT_SIDE = 0;
+    public static final int ZERO = 0;
+    public static final int INDEX_TO_BE_DELETED = 1;
```
Comment (OzairHasan) [inline src/main/java/seedu/allonus/expense/ExpenseParser.java:17]: Good use of named constants.

**Comment by mathanmahe** [comment]
I've linked an issue that I think this PR would fix - Issue #41

File: src/main/java/seedu/allonus/expense/ExpenseParser.java [hunk 2]
```diff
@@ -22,27 +38,48 @@
...
+     * @return the contents of the field specified by the delimiter
+     * @throws IndexOutOfBoundsException if contents supplied is missing
+     */
     public static String parseKeywordExpense(String userInput, String leftDelimiter, String rightDelimiter)
             throws IndexOutOfBoundsException {
-        String[] stripLeftOfDelimiter = userInput.split(leftDelimiter, 2);
+        assert userInput != null : ASSERT_INPUT_NOT_NULL;
+        assert leftDelimiter != null : ASSERT_DELIMITER_NOT_NULL;
+        assert rightDelimiter != null : ASSERT_DELIMITER_NOT_NULL;
+        String[] stripLeftOfDelimiter = userInput.split(leftDelimiter, SPLIT_IN_HALF);
```
Comment (OzairHasan) [inline src/main/java/seedu/allonus/expense/ExpenseParser.java:54]: Good use of assertions to check assumptions.

**Review by OzairHasan — COMMENTED** [review]
LGTM

**Pull request merged**